### PR TITLE
feat: wip icons refactor

### PR DESCRIPTION
**Pull request by mathisscott**
• created cwc-icon
• aliased clr-icon so it can also be used
• identity helper functions
• bare-bones api
• brought over singleton logic from old icons api (seems to work fine)
• added interfaces
• added shape getter to icon element
• adding composable css classname functionality
• api tests and functionality
• testing renderer to ensure only one call per icon
• rewrote renderer
• separated renderer helpers into their own file

Signed-off-by: Scott Mathis <smathis@vmware.com>

## PR Checklist

Please check if your PR fulfills the following requirements:

* [ ] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

* [ ] Bugfix
* [ ] Feature
* [ ] Code style update (formatting, local variables)
* [ ] Refactoring (no functional changes, no api changes)
* [ ] Build related changes
* [ ] CI related changes
* [ ] Documentation content changes
* [ ] clarity.design website / infrastructure changes
* [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

* [ ] Yes
* [ ] No

### CODE REVIEW

File: golden/clr-core.d.ts [hunk 1]
```diff
@@ -1,10 +1,30 @@
...
+
 export declare class CwcButton extends LitElement {
     render(): import("lit-element").TemplateResult;
     static readonly styles: import("lit-element").CSSResult;
 }
 
 export declare class CwcIcon extends LitElement {
+    _shape: string;
+    _size: string | number;
+    shape: string;
```
Comment (jeeyun) [inline golden/clr-core.d.ts]: So we have `shape` and `_shape`?

Comment (mathisscott) [reply golden/clr-core.d.ts]: Yep. It's the same sort of thing we do in Clarity Angular. `shape` is a getter. People could pass in anything. So the getter does a lookup. If the shape is in the registry, then we show it. If not, we display the `unknown` icon.

File: src/clr-core/common/utils/register.spec.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+/* jasmine is running these async and it's yielding weird passes/fails */
+// import { registerElementSafely } from './register';
+// import { elementExists } from './exists';
```
Comment (jeeyun) [inline src/clr-core/common/utils/register.spec.ts]: This whole test file is commented out. Were there issues with running the tests?

File: src/dev-core/src/app/dropdown/dropdown.demo.ts [hunk 3]
```diff
@@ -4,8 +4,15 @@
...
  */
 import { Component } from '@angular/core';
+import '@clr/core/icon';
+import { ClarityIcons } from '@clr/core/icon/icon.service';
+import { oddballIcon } from '@clr/core/icon/shapes/oddball';
 import '@clr/core/test-dropdown';
 
+ClarityIcons.add(['oddball', oddballIcon]);
+
+console.log(ClarityIcons.registry);
```
Comment (jeeyun) [inline src/dev-core/src/app/dropdown/dropdown.demo.ts]: cleanup: remove console log?

Comment (mathisscott) [reply src/dev-core/src/app/dropdown/dropdown.demo.ts]: Yep

File: src/clr-core/common/utils/register.spec.ts (hunk 2 above)
Comment (mathisscott) [reply src/clr-core/common/utils/register.spec.ts]: Yes. I'll need to get with Cory about this. the comment on line 7 explains what i was seeing.

Comment (mathisscott) [reply src/clr-core/common/utils/register.spec.ts]: Yep. I'll need to get with Cory about this. Line 7 explains a bit about what i was seeing.

**Pull request closed without merging**